include/zephyr/bluetooth/audio/pbp.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_PBP_
+#define ZEPHYR_INCLUDE_BLUETOOTH_AUDIO_PBP_
+
+/**
+ * @brief Public Broadcast Profile (PBP)
+ *
+ * @defgroup bt_pbp Public Broadcast Profile (PBP)
+ *
+ * @ingroup bluetooth
+ * @{
+ *
+ * [Experimental] Users should note that the APIs can change
+ * as a part of ongoing development.
+ */
+
+#include <zephyr/sys/util.h>
+
+#include <zephyr/bluetooth/audio/audio.h>
+
+/** PBP Roles */
+enum bt_pbp_role {
+	BT_PBP_ROLE_PBS = BIT(0), /*< Public Broadcast Source */
+	BT_PBP_ROLE_PBK = BIT(1), /*< Public Broadcast Sink */
+	BT_PBP_ROLE_PBA = BIT(2), /*< Public Broadcast Assistant */
+};
+
+/** Broadcast Stream encryption state */
+enum bt_pbp_encryption {
+	BT_PBP_STREAMS_NOT_ENCRYPTED = 0,
+	BT_PBP_STREAMS_ENCRYPTED = 1,
+};
+
+/** Broadcast Stream Quality */
+enum bt_pbp_quality {
+	BT_PBP_AUDIO_CONFIG_NOT_PRESENT = 0,
+	BT_PBP_AUDIO_CONFIG_PRESENT = 1,
+};
+
+/** Public Broadcast Announcement features */
+enum bt_pbp_announcement_features {
+	BT_PBP_ANNOUNCEMENT_FEATURE_ENCRYPTION = BIT(0),
+	BT_PBP_ANNOUNCEMENT_FEATURE_STANDARD_QUALITY = BIT(1),
+	BT_PBP_ANNOUNCEMENT_FEATURE_HIGH_QUALITY = BIT(2),
+};
+
+/**
+ * @brief Sets the received PBP role(s) and Public Broadcast Announcement features.
+ *
+ * @param role		PBP role(s) of the device (one or multiple).
+ * @param pba_params	Public Broadcast Announcement features
+ *
+ */
+void bt_pbp_init_source(enum bt_pbp_role role, uint8_t pba_params);
+
+/**
+ * @brief Sets the received PBP role(s).
+ *
+ * @param role		PBP role(s) of the device (one or multiple).
+ *
+ */
+void bt_pbp_set_role(enum bt_pbp_role role);
+
+/**
+ * @brief Creates a Public Broadcast Announcement based on the information received
+ * in bt_pbp_init_source().
+ *
+ * @param meta_count	Number of metadata fields to be included in the advertising data
+ * @param meta		Metadata to be included in the advertising data
+ * @param pba_adv_data	Advertising data structure to include the Public Broadcast Announcement
+ *
+ * @return	0 on success or an appropriate error code.
+ */
+int bt_pbp_create_public_broadcast_announcement(size_t meta_count,
+			const struct bt_audio_codec_data meta[], struct bt_data *pba_adv_data);
+
+/**
+ * @brief Parses the received advertising data and looks for the Public Broadcast
+ * Announcement Service UUID. If found, it checks the received stream configuration
+ * preferences against the ones advertised by the source.
+ *
+ * @param ad				Advertising data to be checked
+ * @param sink_config_preferences	Sink stream configuration preferences
+ *
+ * @return true if the stream fits the sink's preferences, false otherwise
+ */
+bool bt_pbp_parse_public_broadcast_announcement(struct bt_data *ad,
+			uint8_t sink_config_preferences);
```

Comment:
Did you consider just to parse and return the parsed result (that's what the name of this function implies)?
E.g.
```suggestion
  * @param ad[in]                Advertising data as given to the bt_le_scan_cb_t callback
  * @param features[out]   PBA features  
  * @param meta[out]		PBA metadata
int bt_pbp_parse_public_broadcast_announcement(struct net_buf_simple *ad, enum bt_pbp_announcement_features *features, struct bt_data *meta_data);
```

Reply:
If we keep this function (which I am also unsure about), then I agree with @MariuszSkamra . This function does not implement any behavior stated by the profile, but may be useful to extract the data. 

Once https://github.com/zephyrproject-rtos/zephyr/pull/58079 is merged, then the metadata can just be returned as a pointer, as the current structure for meta and config data will become flat arrays.